openid/src/main/java/com/inrupt/client/openid/OpenIdSession.java

```diff
@@ -225,7 +225,7 @@ public Optional<Credential> fromCache(final Request request) {
         final Credential c = credential.get();
         if (!hasExpired(c) && request != null && requestCache.get(cacheKey(request.uri())) != null) {
             LOGGER.debug("Using cached token for request: {}", request.uri());
-            return Optional.of(c);
+            return Optional.of(requestCache.get(cacheKey(request.uri())));
```

Comment:
We'll want to check whether this cached credential has expired. This function would look something like:

```java
if (request != null) {
    final Credential c = requestCache.get(cacheKey(request.uri()));
    if (!hasExpired(c)) {
        return Optional.of(c);
    }
}
return Optional.empty();
```

Reply:
Fixed in 6c3f360699f6e8711c163efa3ac87948a685cd63